Список фильмов переведен на RecycledView

demqn83 opened the pull request:
В описании фильмов список актеров переведен на RecycledView
![Screenshot_20201205_191053](https://user-images.githubusercontent.com/73488383/101249445-4e808180-372e-11eb-91b4-d0c26a2d3016.png)
![Screenshot_20201205_191126](https://user-images.githubusercontent.com/73488383/101249564-5809e980-372e-11eb-9f44-6af3cec80eb0.png)

demqn83 committed:
Список фильмов переведен на RecycledView
В описании фильмов список актеров переведен на RecycledView

demqn83 committed:
Изменен манифест для работы на 30 API.
Работа лайка реализована через notifyItemChanged()

ivanlardis commented on app/src/main/AndroidManifest.xml:
```diff
@@ -23,4 +24,7 @@
             android:resource="@array/preloaded_fonts" />
     </application>
 
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
```
ACCESS_NETWORK_STATE а это точно нужно?

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -20,14 +25,45 @@ class FragmentMoviesDetails : Fragment() {
+        val position = arguments?.getInt("position")
         val view = inflater.inflate(R.layout.fragment_movies_details, container, false)
+
+        val nameMovie: TextView? = view.findViewById(R.id.film_name_text_view)
```
а зачем nullable?

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -20,14 +25,45 @@ class FragmentMoviesDetails : Fragment() {
+
+        val movie = FakeMovies().getMoviesById(position!!)
+        nameMovie?.text = movie.nameMovie
+        reviews?.text = "${movie.reviews} reviews"
```
dнужно брать из ресурсов слово " reviews"

demqn83 replied:
Получение ресурсов переделал.

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -20,14 +25,45 @@ class FragmentMoviesDetails : Fragment() {
                               savedInstanceState: Bundle?): View? {
+
+        arguments = getArguments();
+        val position = arguments?.getInt("position")
```
фрагменты создаем через newInstance

и position в приватной константе
и должна быть не позиция, а id ролика
![image](https://user-images.githubusercontent.com/11761610/101289124-97f2ce80-380b-11eb-8a5b-5101f69e6574.png)

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesDetails.kt:
```diff
@@ -20,14 +25,45 @@ class FragmentMoviesDetails : Fragment() {
+        val actors = FakeMovies().getMoviesById(position).listOfActors
+        val adapter = ActorsAdapter(container!!.context, actors)
+        list.adapter = adapter
+        list.layoutManager = LinearLayoutManager(container!!.context, LinearLayoutManager.HORIZONTAL, false);
```
У фрагмента есть доступ к активити и контексту
 
container!!.context ->requireContext()

ivanlardis commented on app/src/main/java/ru/demqn/appname/ActorsAdapter.kt:
```diff
@@ -0,0 +1,42 @@
+
+class ActorsViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
+    private val nameActor: TextView? = itemView.findViewById(R.id.actor_name_text_view)
+    private val photoActor: ImageView? = itemView.findViewById(R.id.actor_photo_image_view)
```
зачем nullable?

ivanlardis commented on app/src/main/java/ru/demqn/appname/ActorsAdapter.kt:
```diff
@@ -0,0 +1,42 @@
+        Glide
+            .with(itemView.context)
+            .load(actor.photoActor)
+            .override(80, 80)
```
тут задаются значения в pixels 
а в разметке в dp
лучше пока удалить либо переводить

ivanlardis commented on app/src/main/java/ru/demqn/appname/ActorsAdapter.kt:
```diff
@@ -0,0 +1,42 @@
+    private val nameActor: TextView? = itemView.findViewById(R.id.actor_name_text_view)
+    private val photoActor: ImageView? = itemView.findViewById(R.id.actor_photo_image_view)
+
+    fun Bind(actor: Actor) {
```
названия метода должно быть с маленькой буквы

ivanlardis commented on app/src/main/java/ru/demqn/appname/ActorsAdapter.kt:
```diff
@@ -0,0 +1,42 @@
+
+    private val inflater : LayoutInflater = LayoutInflater.from(context)
+
+    fun getItem(position: Int) : Actor = actors[position]
```
) : A реформат кода нужен

ivanlardis commented on app/src/main/java/ru/demqn/appname/ActorsAdapter.kt:
```diff
@@ -0,0 +1,42 @@
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+
+class ActorsAdapter(context: Context, var actors:List<Actor>) : RecyclerView.Adapter<ActorsViewHolder>(){
```
Так было в воркшопе
не ныжно передавать контекст через конструктор
![image](https://user-images.githubusercontent.com/11761610/101289271-7a723480-380c-11eb-9860-bc76aff14e81.png)

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesList.kt:
```diff
@@ -5,49 +5,39 @@ import android.os.Bundle
 
 
-class FragmentMoviesList : Fragment() {
+class FragmentMoviesList() : Fragment() {
```
убрать ()
они не нужне если нет параметров в конструкторе

ivanlardis commented on app/src/main/java/ru/demqn/appname/FragmentMoviesList.kt:
```diff
@@ -5,49 +5,39 @@ import android.os.Bundle
 
     private var listener: TransactionsFragmentClicks? = null
     private var colorLike:Boolean = false
+    var adapterList:MoviesAdapter? = null
```
долен быть приватным
и оброботку клика на like
не нужно переносить в активити
лучше пока оставить тут
А потом доделаем с архитектурой

ivanlardis commented on app/src/main/java/ru/demqn/appname/MainActivity.kt:
```diff
@@ -27,9 +26,11 @@ class MainActivity : AppCompatActivity(), FragmentMoviesList.TransactionsFragmen
-
+    override fun addMovieDetails(position:Int) {
+        val bundle = Bundle()
+        bundle.putInt("position", position)
```
создание и засечивание данных перенесется
в метод newInstance фрагмента

ivanlardis commented on app/src/main/java/ru/demqn/appname/MainActivity.kt:
```diff
@@ -38,6 +39,12 @@ class MainActivity : AppCompatActivity(), FragmentMoviesList.TransactionsFragmen
         }
     }
 
+    override fun clickLike(position: Int) {
```
Должно Обрабатыватся внутри фрагмента
в активити не выносим эту логику

ivanlardis commented on app/src/main/java/ru/demqn/appname/Movie.kt:
```diff
@@ -0,0 +1,16 @@
+        val rating: Double,
+        val movieGenre: String,
+        val rated: String,
+        var like: Boolean,
```
like делаем val
и используем метод copy как писал в телеграме

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesAdapter.kt:
```diff
@@ -0,0 +1,74 @@
+        holder.Bind(getItem(position), position)
+    }
+
+    override fun getItemCount(): Int = FakeMovies().getListMovies().size
```
в адаптер должен сетится лист с даными из фрагмента
а фрагмент уже берет из  FakeMovies()

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesAdapter.kt:
```diff
@@ -0,0 +1,74 @@
+class DataViewHolder(itemView: View, private var listener: FragmentMoviesList.TransactionsFragmentClicks) : RecyclerView.ViewHolder(itemView){
+    private val nameMovie: TextView? = itemView.findViewById(R.id.nameMovie_text_view)
+    private val movieDuration: TextView? = itemView.findViewById(R.id.film_duration_text_view)
+    private val reviews: TextView? = itemView.findViewById(R.id.description_rating_text_view)
```
удаляем ?- nullable

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesAdapter.kt:
```diff
@@ -0,0 +1,74 @@
+    private var like: ImageView? = itemView.findViewById(R.id.ic_like_image_view)
+    private val poster: ImageView? = itemView.findViewById(R.id.poster_image_view)
+
+    fun Bind(movie: Movie, position:Int) {
```
Bind ->bind

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesAdapter.kt:
```diff
@@ -0,0 +1,74 @@
+//            .placeholder(R.drawable.chris_hemsworth)
+            .into(poster!!);
+
+        if (movie.like) {
```
![image](https://user-images.githubusercontent.com/11761610/101291325-f96d6a00-3818-11eb-9ff7-1ef756841326.png)

можно так

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesAdapter.kt:
```diff
@@ -0,0 +1,74 @@
+            listener.clickLike(position)
+        }
+
+        itemView.findViewById<ImageView>(R.id.ic_bg_image_view).setOnClickListener {
```
почему ты ее не получешь там же где получаешь все остальные
![image](https://user-images.githubusercontent.com/11761610/101291341-2c176280-3819-11eb-949b-2f53b03ee5b9.png)

ivanlardis commented on app/src/main/java/ru/demqn/appname/MoviesDataSource.kt:
```diff
@@ -0,0 +1,322 @@
+package ru.demqn.appname
+
+class MoviesDataSource {
```
Зачем этот класс если его не используем?

demqn83 committed:
Убраны лишние null типы.
Создание фрагментов сделано на newInstance.
Из конструктора адаптеров убран контекст.
Сделано получение строковых реквизитов из ресурсов.

demqn83 committed:
Переделаны обработчики кликов для списка фильмов.

demqn83 committed:
Обновление в списке конкретного фильма.

demqn83 committed:
Получение рейтинга из ресурсов.
